src/main/ruby/truffleruby/core/string.rb

```diff
@@ -70,6 +70,49 @@ def byteslice(index_or_range, length = undefined)
     byteslice index, length
   end
 
+  def bytesplice(index_or_range, length = undefined, str)
+    is_range = Primitive.is_a?(index_or_range, Range)
+
+    if Primitive.undefined?(length)
+      raise TypeError, "wrong argument type #{Primitive.class(index_or_range)} (expected Range)" unless is_range
+
+      start, len = Primitive.range_normalized_start_length(index_or_range, bytesize)
+      len = 0 if len < 0
+    else
+      start = Primitive.rb_to_int(index_or_range)
+      start += bytesize if start < 0
+      len = Primitive.rb_to_int(length)
+    end
+
+    str = StringValue(str)
+
+    if len < 0
+      raise IndexError, "negative length #{len}"
+    end
+
+    if bytesize < start || start < 0
+      if is_range
+        raise RangeError, "#{index_or_range} out of range"
+      else
+        raise IndexError, "index #{index_or_range} out of string"
+      end
+    end
+
+    len = bytesize - start if len > bytesize - start
```

Comment:
`Primitive.min` would be nicer here, but let's do that in https://github.com/oracle/truffleruby/pull/3043 since this PR is in the merge queue

Reply:
Fixed there.